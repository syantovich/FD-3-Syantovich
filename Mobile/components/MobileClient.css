div.MobileClient
{
  padding: 2px 0;
}

span.MobileClientBalance
{
  float: right;
  width: 30px;
  font-weight: bold;
  text-align: right;
}

td.MobileClientFIO,th
{
  width: 150px;
  margin: 30px;
  padding: 0px;
  text-align: left;
}
td.active{
  background-color: green;
}
td.blocked{
  background-color: red;
}