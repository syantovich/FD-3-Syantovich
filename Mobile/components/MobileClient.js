import React from 'react';
import PropTypes from 'prop-types';
import {voteEvents} from "./events";



class MobileClient extends React.PureComponent {

  static propTypes = {
    client:PropTypes.shape({
      id: PropTypes.number.isRequired,
      FIO:PropTypes.shape({
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
      }),
      balance: PropTypes.number.isRequired,
    })

  };

  state = {
    status:(this.props.client.balance<0)?"blocked":"active",
  };
  UNSAFE_componentWillReceiveProps=(nextProps)=> {
this.setState({status:(nextProps.client.balance<0)?"blocked":"active",})
}

  render() {

    console.log("MobileClient id="+this.props.client.id+" render");
    
    return (
        <tbody>
      <tr className='MobileClient'>

        <td className='MobileClientFIO'>{this.props.client.im}</td>
        <td className='MobileClientFIO'>{this.props.client.fam}</td>
        <td className='MobileClientFIO'>{this.props.client.otch}</td>
        <td className='MobileClientFIO'>{this.props.client.balance}</td>
        <td className={'MobileClientFIO '+this.state.status}>{this.state.status}</td>
        <td className='MobileClientFIO'><input type="button" value="Edit" onClick={()=>{voteEvents.emit("EsetMode","Edit",this.props.client.id)}}/></td>
        <td className='MobileClientFIO'><input type="button" value="Delete" onClick={()=>{voteEvents.emit("Edelete",this.props.client.id)}}/></td>
      </tr>
        </tbody>
    );

  }

}

export default MobileClient;
