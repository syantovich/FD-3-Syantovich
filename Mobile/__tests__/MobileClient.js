import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from "../components/MobileClient";

test('работа MobileClient', () => {

    const client = { id: 101, FIO: { fam: "Иванов", im: "Иван", otch: "Иванович" }, balance: 200 };

    const component = renderer.create(
        <MobileClient client={client} />
    );

    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const buttonEdit = component.root.findByProps({className:"Edit101"})
    const buttonDelete = component.root.findByProps({className:"Delete101"})


    buttonEdit.props.onClick();
    buttonDelete.props.onClick();

    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


    buttonEdit.props.onClick();
    buttonDelete.props.onClick();

    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


});