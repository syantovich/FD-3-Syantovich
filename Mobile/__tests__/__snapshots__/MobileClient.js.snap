// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileClient 1`] = `
<tbody>
  <tr
    className="MobileClient"
  >
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    >
      200
    </td>
    <td
      className="MobileClientFIO active"
    >
      active
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Edit101"
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Delete101"
        onClick={[Function]}
        type="button"
        value="Delete"
      />
    </td>
  </tr>
</tbody>
`;

exports[`работа MobileClient 2`] = `
<tbody>
  <tr
    className="MobileClient"
  >
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    >
      200
    </td>
    <td
      className="MobileClientFIO active"
    >
      active
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Edit101"
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Delete101"
        onClick={[Function]}
        type="button"
        value="Delete"
      />
    </td>
  </tr>
</tbody>
`;

exports[`работа MobileClient 3`] = `
<tbody>
  <tr
    className="MobileClient"
  >
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    />
    <td
      className="MobileClientFIO"
    >
      200
    </td>
    <td
      className="MobileClientFIO active"
    >
      active
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Edit101"
        onClick={[Function]}
        type="button"
        value="Edit"
      />
    </td>
    <td
      className="MobileClientFIO"
    >
      <input
        className="Delete101"
        onClick={[Function]}
        type="button"
        value="Delete"
      />
    </td>
  </tr>
</tbody>
`;
